### ISSUE TITLE
BUG route current is true for every substr of routename

### ISSUE THREAD

**Issue by thoresuenert** [tightenco/ziggy#213]
## For bug reports only please provide
the regex build by `route().current('events.*')` is wrong.
Given from the test case, the following assertion it correct:
```javascript
assert.equal(
  "events.venues.show",
  route().current()
);
```
From my understanding the following assertions should be correct as well:
```javascript
assert.equal(
  false,
  route().current("events")
);
assert.equal(
  true,
  route().current("events.*")
);
assert.equal(
  false,
  route().current("show")
);
```

but they are incorrect.

We found out that the regex in this line https://github.com/tightenco/ziggy/blob/253b919cddc196a5a9a94c5d1e406d1b57f700f1/src/js/route.js#L131 must include starting and ending like this:
```js 
const pattern = new RegExp('^' + name.replace('*', '.*').replace('.', '\.') + '$', 'i');
```
### Currently installed Laravel version:
"version": "v5.7.28",

### Currently installed Ziggy version
"version": "v0.6.9",

### Ziggy call in context
Here is my real world example:
```javascript
route().current()
  "job.detail.messages"
route().current("message")
  true
route().current("ob")
  true
route().current("detail")
  true
route().current("tail")
  true
```

**Comment by nmsmith22389** [comment]
Thanks for finding this! 👍 I could see this really causing some unwanted side effects. Hope the PR gets merged soon!

**Comment by jakebathman** [comment]
Closing this, since the PR was merged as #235 (had to be re-done, it's a GitHub issue). Thank you @thoresuenert! 🎉
